### ISSUE TITLE
Unable to cover .NET 4.0 web application 

### ISSUE THREAD

**Issue by danielhuber3008** [SteveGilham/altcover#38]
I am trying to use altcover for the coverage measurement of a .NET 4.0 web application deployed on IIS 8.5.

Therefore, I execute the following command:
`altcover --save --inplace -i $(OutputDir)`

where `$(OutputDir)` is the bin folder of the desired website.

Besides the instrumend assemblies, this results in two additional files in the bin folder: _AltCover.Recorder.g.dll_ and _AltCover.Recorder.g.pdb._

After restarting the w3wp.exe, the website fails with a runtime error.

The Event Viewer retrieves the following error: 
```
Exception information: 
Exception type: ConfigurationErrorsException 
Exception message: Could not load file or assembly 'AltCover.Recorder.g, Version=4.0.0.0, Culture=neutral, PublicKeyToken=4ebffcaabf10ce6a' or one of its dependencies. Strong name signature could not be verified. The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)
at System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective)
at System.Web.Configuration.CompilationSection.LoadAllAssembliesFromAppDomainBinDirectory()
at System.Web.Configuration.CompilationSection.LoadAssembly(AssemblyInfo ai)
at System.Web.Compilation.BuildManager.GetReferencedAssemblies(CompilationSection compConfig)
at System.Web.Compilation.BuildManager.GetPreStartInitMethodsFromReferencedAssemblies()
at System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded)
at System.Web.Compilation.BuildManager.ExecutePreAppStart()
at System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException)
```

What am I doing wrong? Is my approach possible/correct? 
I should also add that modifying the web application **before** the build is not possible. 

Thank you!

**Comment by SteveGilham** [comment]
The approach you are taking is the correct one; and the instrumentation has to be done post-build in any case.  

A couple of items for clarification 
1. when you write 
```
altcover --save --inplace -i $(OutputDir)
```
are you using the global tool version of AltCover here or the .net Framework executable?
2. is any of the code you build also strong-named?

**Comment by SteveGilham** [comment]
I have repro'd this with the case of 

1. the .net Framework tool
2. no strong-naming

**Comment by SteveGilham** [comment]
There is a work-round, by using the .net core version of the tool.

This will place the instrumentation recorder files AltCover.Recorder.g.dll and AltCover.Recorder.g.pdb in the output, along with a copy of FSharp.Core.dll upon which the recorder assembly relies (in the .net framework version that is static-linked into the assembly, but that trick doesn't work for .net core, or at least didn't last I tried).

**Comment by SteveGilham** [comment]
Should be fixed in release [4.0.649](https://www.nuget.org/packages/AltCover/4.0.649).

**Comment by danielhuber3008** [comment]
For clarification - yes, I used the the .net Framework tool and no strong-naming. 
The new release fixed this issue - thanks a lot! 👍
